packages/api/amplify_api/lib/amplify_api.dart

```diff
@@ -1,51 +1,20 @@
-/*
- * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
- *
- * Licensed under the Apache License, Version 2.0 (the "License").
- * You may not use this file except in compliance with the License.
- * A copy of the License is located at
- *
- *  http://aws.amazon.com/apache2.0
- *
- * or in the "license" file accompanying this file. This file is distributed
- * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
- * express or implied. See the License for the specific language governing
- * permissions and limitations under the License.
- */
-
-library amplify_api_plugin;
-
-import 'dart:io';
-
-import 'package:amplify_api/src/method_channel_api.dart';
-import 'package:amplify_core/amplify_core.dart';
-import 'package:meta/meta.dart';
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
 
 export 'package:amplify_core/src/types/api/api_types.dart';
 
-export './model_mutations.dart';
-export './model_queries.dart';
-export './model_subscriptions.dart';
-
-/// {@template amplify_api.amplify_api}
-/// The AWS implementation of the Amplify API category.
-/// {@endtemplate}
-abstract class AmplifyAPI extends APIPluginInterface {
```

Comment:
So, since we're not planning to ship this for a while, can we leave the method channel stuff in place for iOS/Android and use the Dart impl for just Web/Desktop?

Basically you could use this class and just replace the `UnsupportedError` with a return of the Dart plugin